File: sparkcql/src/test/scala/gov/va/sparkcql/common/StopwatchTest.scala
package gov.va.sparkcql.common

import org.scalatest.flatspec.AnyFlatSpec

class StopwatchTest extends AnyFlatSpec {
  "A Stopwatch" should "calculation duration" in {
    var watch = Stopwatch()
    Thread.sleep(10)
    watch.stop()
    assert(watch.lapse.toMillis >= 10)
  }

  it should "format properly" in {
    var watch = Stopwatch()
    Thread.sleep(100)
    watch.stop()
    assert(watch.toString.startsWith("0h:0m:0s:1") && watch.toString.endsWith("ms"))
    assert(1 == 1)
  }
}

File: sparkcql-core/src/test/java/gov/va/sparkcql/common/StopwatchTest.java
package gov.va.sparkcql.common;

import static org.junit.jupiter.api.Assertions.assertTrue;
import org.junit.jupiter.api.Test;

public class StopwatchTest {

    @Test
    public void should_calculate_duration() {
        var watch = new Stopwatch();
        watch.sleep(10);
        watch.stop();
        assertTrue(watch.getLapse().toMillis() >= 10);
    }

    @Test
    public void should_format_properly() {
        var watch = new Stopwatch();
        watch.sleep(100);
        watch.stop();
        assertTrue(watch.toString().startsWith("0h:0m:0s:") && watch.toString().endsWith("ms"));
    }
}